### PR TITLE
Fix: Use `importlib.resources` or `importlib_resources` instead of `pkg_resources`

### LINKED ISSUE #11448: `mne` depends on `setuptools` without declaring it as a dependecy, which can break build backends other than `setuptools`
### Description of the problem

Without declaring `setuptools` as a dependency, `mne` imports `pkg_resources`, which is a part of `setuptools`, in at least one place:

```
File "/tmp/pip-build-env-t3o1ln7x/normal/lib/python3.9/site-packages/mne/datasets/eegbci/eegbci.py", line 8, in <module>
          import pkg_resources
```

In my case this leads to the following problem: I have written a package that depends on `mne` and that I package with `flit`, i.e. my package is fully independent of `setuptools`. When I try to install my package (with the `flit_core` build backend instead of `setuptools`) I get the following error:

```
pip install .[test]
Processing /home/pzr
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [38 lines of output]
      Unexpected names under [project]: requires_python
      [pyhelpers] SSL verification check: SSL verification failed.
      [pyhelpers] Adding FhG certificate bundle to the certificate store.
      [pyhelpers] Checking again: SSL verification works, continuing.
      Traceback (most recent call last):
        File "/usr/local/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
          main()
        File "/usr/local/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/usr/local/lib/python3.9/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 164, in prepare_metadata_for_build_wheel
          return hook(metadata_directory, config_settings)
        File "/tmp/pip-build-env-t3o1ln7x/overlay/lib/python3.9/site-packages/flit_core/buildapi.py", line 49, in prepare_metadata_for_build_wheel
          metadata = make_metadata(module, ini_info)
        File "/tmp/pip-build-env-t3o1ln7x/overlay/lib/python3.9/site-packages/flit_core/common.py", line 410, in make_metadata
          md_dict.update(get_info_from_module(module, ini_info.dynamic_metadata))
        File "/tmp/pip-build-env-t3o1ln7x/overlay/lib/python3.9/site-packages/flit_core/common.py", line 207, in get_info_from_module
          docstring, version = get_docstring_and_version_via_import(target)
        File "/tmp/pip-build-env-t3o1ln7x/overlay/lib/python3.9/site-packages/flit_core/common.py", line 180, in get_docstring_and_version_via_import
          spec.loader.exec_module(m)
        File "<frozen importlib._bootstrap_external>", line 850, in exec_module
        File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
        File "/home/pzr/src/pyhelpers/__init__.py", line 3, in <module>
          from . import constants, fileio, neuro, plotting, sigproc, string
        File "/home/pzr/src/pyhelpers/sigproc/__init__.py", line 1, in <module>
          from . import eeg
        File "/home/pzr/src/pyhelpers/sigproc/eeg/__init__.py", line 1, in <module>
          from ._eeg import (create_eeg_raw_array, drop_channels, filter_eeg,
        File "/home/pzr/src/pyhelpers/sigproc/eeg/_eeg.py", line 4, in <module>
          from mne import create_info
        File "/tmp/pip-build-env-t3o1ln7x/normal/lib/python3.9/site-packages/mne/__init__.py", line 111, in <module>
          from . import datasets
        File "/tmp/pip-build-env-t3o1ln7x/normal/lib/python3.9/site-packages/mne/datasets/__init__.py", line 10, in <module>
          from . import eegbci
        File "/tmp/pip-build-env-t3o1ln7x/normal/lib/python3.9/site-packages/mne/datasets/eegbci/__init__.py", line 3, in <module>
          from .eegbci import data_path, load_data, standardize
        File "/tmp/pip-build-env-t3o1ln7x/normal/lib/python3.9/site-packages/mne/datasets/eegbci/eegbci.py", line 8, in <module>
          import pkg_resources
      ModuleNotFoundError: No module named 'pkg_resources'
      [end of output]
```

The problem is solved by declaring `setuptools` as a dependency in my own package, which I would prefer not having to do for obvious reasons. I understand that `setuptools` has been the de-facto standard for many years and that one could just assume that pretty much every package will be installed in an environment where it is present. But lately a lot of people have been moving away from `setuptools`, so if you depend on it I think you should explicitly declare it as a dependency in `mne`.

### Steps to reproduce

Write a package with a build backend other than `setuptools` and have it depend on `mne`.

### Link to data

_No response_

### Expected results

Package can be installed with build backend other than `setuptools`.

### Actual results

Package can't be installed.

### Additional information

(This is on a Gitlab runner, so not sure how helpful this is)

```
Platform:         Linux-3.10.0-1160.15.2.el7.x86_64-x86_64-with-glibc2.31
Python:           3.9.16 (main, Jan 23 2023, 23:35:25)  [GCC 10.2.1 20210110]
Executable:       /usr/local/bin/python
CPU:              : 20 cores
Memory:           Unavailable (requires "psutil" package)
mne:              1.3.0
numpy:            1.24.1 {unknown linalg bindings (threadpoolctl module not found: No module named 'threadpoolctl')}
scipy:            1.10.0
matplotlib:       3.6.3 {backend=agg}

sklearn:          Not found
numba:            Not found
nibabel:          Not found
nilearn:          Not found
dipy:             Not found
openmeeg:         Not found
cupy:             Not found
pandas:           Not found
pyvista:          Not found
pyvistaqt:        Not found
ipyvtklink:       Not found
vtk:              Not found
qtpy:             Not found
ipympl:           Not found
pyqtgraph:        Not found
pooch:            v1.6.0

mne_bids:         Not found
mne_nirs:         Not found
mne_features:     Not found
mne_qt_browser:   Not found
mne_connectivity: Not found
mne_icalabel:     Not found
```

### PR DESCRIPTION
**Pull request by aplzr**
#### Reference issue
Fixes #11448.

#### What does this implement/fix?
`mne` currently relies on `pkg_resources`, which is a part of `setuptools`, without declaring it as a dependency. This can cause issues in environments where `setuptools` is not present by default (see #11448 for an example). 

This PR updates `mne` to use `importlib.resources` for Python >= 3.9 or its backport `importlib_resources` for Python < 3.9 as recommended [here](https://setuptools.pypa.io/en/latest/pkg_resources.html). `importlib` is part of the standard library.

#### Additional information
The only `importlib` function this PR uses is `importlib.resources.files()`, which was added in Python 3.9. I added a dependency to `importlib_resources` in `requirements_base.txt` for Python versions < 3.9. Please advise if this is ok. If not I will update the PR accordingly, but it would mean that `mne` would still rely on `pkg_resources` for Python < 3.9 (meaning that `setuptools` should be declared as a dependency then).

`pkg_resources` was used in two functions, `mne/datasets/eegbci/eegbci.py:load_data()` and `mne/utils/misc.py:_resource_path()`. There are currently no unit tests that test `load_data()`, but manually calling the updated version works for Python 3.7 and 3.11 on my end. All unit tests calling the updated `_resource_path()` are passing when I run them locally in Python 3.7 and 3.11.

### CODE REVIEW

**Comment by cbrnr** [comment]
Regarding the use of checksums in `eegbci`, I wonder if it's really necessary to query a local file with `importlib`. Could we not fetch the [original checksum file](https://physionet.org/files/eegmmidb/1.0.0/SHA256SUMS.txt) and check against its contents? I feel like it's probably not worth having this mechanism just for our checksums.

**Review by larsoner — APPROVED** [review]
> I feel like it's probably not worth having this mechanism just for our checksums.

Agreed but we do make use of `files` in `mne/utils/misc.py` already, so why not keep/use it

Changes LGTM, @drammock feel free to merge if you agree this is a reasonable fix. I was never totally sure about the `packaging` / `pkg_resources` etc. stuff...

File: mne/utils/misc.py [hunk 1]
```diff
@@ -22,6 +22,11 @@
 from ..utils import _check_option, _validate_type
 from ._logging import logger, verbose, warn
 
+try:
+    from importlib.resources import files
+except ImportError:
+    from importlib_resources import files
```
Comment (drammock) [inline mne/utils/misc.py:28]:
```suggestion
# TODO: remove try/except when our min version is py 3.9
try:
    from importlib.resources import files
except ImportError:
    from importlib_resources import files
```

File: mne/datasets/eegbci/eegbci.py [hunk 2]
```diff
@@ -4,13 +4,17 @@
...
 from pathlib import Path
 
-from ..utils import _get_path, _do_path_update
 from ...utils import _url_to_local_path, verbose
+from ..utils import _do_path_update, _get_path
+
+try:
+    from importlib.resources import files
+except ImportError:
+    from importlib_resources import files
```
Comment (drammock) [inline mne/datasets/eegbci/eegbci.py:17]:
```suggestion
# TODO: remove try/except when our min version is py 3.9
try:
    from importlib.resources import files
except ImportError:
    from importlib_resources import files
```

**Review by drammock — COMMENTED** [review]
adding a couple of code comments for future maintainers.  Let's see if the CIs are happy...

**Comment by aplzr** [comment]
> Let's see if the CIs are happy...

No such luck I suppose.

**Comment by larsoner** [comment]
It looks like the failures are an unrelated bug where nipy.org is down, so our docs complain. So I think we're actually okay!

**Comment by drammock** [comment]
only failures are the `nipy` ones, so in it goes!  Thanks @aplzr!

**Pull request merged**